railties/test/application/configuration_test.rb

```diff
@@ -274,6 +274,8 @@ def change
 
       ENV["RAILS_GROUPS"] = "javascripts,stylesheets"
       assert_equal [:default, "test", "javascripts", "stylesheets"], Rails.groups
+    ensure
+      ENV.delete "RAILS_GROUPS"
```

Comment:
We shouldn't `ENV.delete` but restore to whatever it was. We got a helper for that (`set_env` I think?) that is copy pasted in a bunch of places. Would be good to clean that up.